core/src/autogluon/core/pseudolabeling/pseudolabeling.py

```diff
@@ -0,0 +1,54 @@
+import numpy as np
+import pandas as pd
+from autogluon.core.constants import PROBLEM_TYPES_CLASSIFICATION
+
+
+def filter_pseudo(y_pred_proba_og, problem_type,
+                  min_proportion_prob: float = 0.05, max_proportion_prob: float = 0.6,
+                  threshold: float = 0.95, percent_sample: float = 0.3):
+    """
+    Takes in the predicted probabilities of the model and chooses the indices that meet
+    a criteria to incorporate into training data. Criteria is determined by problem_type.
+    If mutliclass or binary will choose all rows with max prob over threshold. For regression
+    chooses 30% of the labeled data randomly.
+
+    Parameters:
+    -----------
+    y_pred_proba_og: The predicted probabilities from the current best model. If problem is
+        'binary' or 'multiclass' then it's Panda series of predictive probs, if it's 'regression'
+        then it's a scalar
+    min_proportion_prob: Minimum percentage of total 'y_pred_proba_og` that is below threshold
+        required to trigger threshold change
+    max_proportion_prob: Maximum percentage of total 'y_pred_proba_og` that is above threshold
+        required to trigger threshold change
+    threshold: The predictive probability that must be exceeded in order to be
+        incorporated into the next round of training (ignored for regression)
+    percent_sample: When problem_type is regression this is percent of pseudo data
```

Comment:
should ideally be called `proportion_sample` since the other variables are called proportions. percent is more vague as to whether only numbers in [0,1] are accepted.